src/ilmm.jl

```diff
@@ -29,7 +29,10 @@ julia> f = ILMM(IndependentMOGP([GP(SEKernel())]), rand(2,2));
 
 julia> latent_f = get_latent_gp(f);
 
-julia> latent_f == IndependentMOGP([GP(SEKernel())])
+julia> latent_f isa IndependentMOGP
```

Comment:
why do we want this change? I find the previous one clearer personally

Reply:
Something was going on with equality comparisons of the composite types I introduced, so I changed the doctests to make sure they were equivalent instead of "equal"

Reply:
hmm, what exactly was going on? is there an issue about this? we might want to fix that instead?

Reply:
Given this
```
struct IndependentMOGP{Tfs<:Vector{<:AbstractGP}} <: AbstractGP
    fs::Tfs
end
function independent_mogp(fs::Vector{<:AbstractGP})
    return IndependentMOGP(fs)
end
```
Should we expect:
```
julia> independent_mogp([GP(KernelFunctions.SEKernel())]) == IndependentMOGP([GP(KernelFunctions.SEKernel())])
true
```
I guess yes, and I am mistaking `isequal`, `==` and `===`

Reply:
This is what made me think composite types couldn't be compared directly.
```
julia> independent_mogp([GP(SEKernel())])
IndependentMOGP{Vector{GP{AbstractGPs.ZeroMean{Float64}, SqExponentialKernel{Distances.Euclidean}}}}(GP{AbstractGPs.ZeroMean{Float64}, SqExponentialKernel{Distances.Euclidean}}[GP{AbstractGPs.ZeroMean{Float64}, SqExponentialKernel{Distances.Euclidean}}(AbstractGPs.ZeroMean{Float64}(), Squared Exponential Kernel (metric = Distances.Euclidean(0.0)))])

julia> IndependentMOGP([GP(SEKernel())])
IndependentMOGP{Vector{GP{AbstractGPs.ZeroMean{Float64}, SqExponentialKernel{Distances.Euclidean}}}}(GP{AbstractGPs.ZeroMean{Float64}, SqExponentialKernel{Distances.Euclidean}}[GP{AbstractGPs.ZeroMean{Float64}, SqExponentialKernel{Distances.Euclidean}}(AbstractGPs.ZeroMean{Float64}(), Squared Exponential Kernel (metric = Distances.Euclidean(0.0)))])

julia> independent_mogp([GP(SEKernel())]) == IndependentMOGP([GP(SEKernel())])
false

julia> isequal(independent_mogp([GP(SEKernel())]), IndependentMOGP([GP(SEKernel())]))
false

julia> independent_mogp([GP(SEKernel())]) === IndependentMOGP([GP(SEKernel())])
false
```

Reply:
Yes, I would expect that. `==` checks for equality of two things (and defaults to `===`), while `===` checks that two things are aliased (i.e. the same object in memory).
So I would expect
```julia
julia> independent_mogp([GP(SEKernel())]) == IndependentMOGP([GP(SEKernel())])
true

julia> independent_mogp([GP(SEKernel())]) === IndependentMOGP([GP(SEKernel())])
false
```

Reply:
In case you are getting
```julia
julia> independent_mogp([GP(SEKernel())]) == IndependentMOGP([GP(SEKernel())])
false
```
we should probably define an equality `==` to compare the values rather than check for `===`, so that this doesn't trip us in other places

Reply:
I feel like we shouldn't have to define our own `==` and that it should work for composite types by default. Maybe I should ask in #julia why this might not be working

Reply:
oh sorry didn't see your second reply - we should define `Base.:(==)(a::MOGP, b::MOGP)`

Reply:
It does not work by default for composite types unfortunately.
see e.g.
```julia
julia> struct MyStruct
           a
       end

julia> a = MyStruct(1.0)
MyStruct(1.0)

julia> b = MyStruct(1.0)
MyStruct(1.0)

julia> @which ==(a, b)
==(x, y) in Base at Base.jl:87
```
which is the default that you can inspect with
```julia
@edit ==(a, b)

# shows
# ==(x, y) = x === y
```

Reply:
So you think its worth adding this by comparing fields and type rather than just explicitly checking in the tests? I definitely can do so, I just feel like I haven't seen an `==` dispatch defined for most composite types I have come across.

Reply:
I don't know, it seems pretty common to me (but maybe we have been working in different packages where conventions differ). e.g. [ChainRulesCore defines `==` for `Tangent` type](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/master/src/differentials/composite.jl#L48-L55), so does [NamedDims](https://github.com/invenia/NamedDims.jl/blob/master/src/functions.jl#L87-L94) for it's wrapper array.

Maybe it's less common in higher level packages. The only downside I can really think of is invalidations, but I don't expect it to be a serious issue.

Reply:
@willtebbutt do you have any thoughts on this? Should equality be defined for an `IndependentMOGP`? If so then also for `Orthogonals`, and `ILMMs`?

Reply:
Yeah -- I don't think one should avoid defining checks for equality, but it's pretty reasonable to ask whether / why you need it. In the wild, are you ever likely to need to check for equality between two IndependentMOGPs, or is this just for testing purposes? If it's the latter, I wouldn't bother to write the code as it's just more code.

Reply:
I can't really see why anyone would want to check between the two, but I could certainly be missing cases of when one might. It was really just for testing a super simple one-line constructor helper function!

Reply:
Tbh, we've not defined the equality operator in AbstractGPs, and it has yet to cause any problems. I would be inclined not to bother, and to test this constuctor in a different manner 🤷

Reply:
Resolving for now then.